superset-frontend/src/SqlLab/components/AceEditorWrapper/index.tsx

```diff
@@ -144,8 +144,10 @@ const AceEditorWrapper = ({
   };
 
   const onChangeText = (text: string) => {
-    setSql(text);
-    onChange(text);
+    if (text !== sql) {
```

Comment:
This change prevents the overwrite the sql by `setQueryEditorAndSaveSqlWithDebounce`